### PR TITLE
Rewrite docker compose

### LINKED ISSUE #11: UPDATE docker compose config

### PR DESCRIPTION
**Pull request by wanglei4687**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #11

**What this PR does / why we need it:**

- Matrixone cluster config update
- Optimized cluster boot mode

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: docker-compose/Makefile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+.PHONY: matrixone production start
```
Comment (sukki37) [inline docker-compose/Makefile:1]: Refactor Advice:
```makefile
PLATFORM ?= linux/amd64
TAG ?= latest
IMAGE ?= matrixorigin/matrixone
BRANCH ?= main

.PHONY: matrixone production start
up:
	export IMAGE=${IMAGE} TAG=${TAG}; docker-compose -f docker-compose.yml up -d

.PHONY: matrixone production stop
down:
	docker-compose -f docker-compose.yml down --remove-orphans

.PHONY: clean data and log
clean:
	rm -rf rm -rf log* && rm -rf store*

.PHONY: clean log, data, repo
all-clean:
	rm -rf log* && rm -rf store* && rm -rf matrixone
 
.PHONY: matrixone dev clone and build
dev-pre:
	git clone git@github.com:matrixorigin/matrixone.git -b ${BRANCH} && cp Dockerfile matrixone
	
.PHONY: matrixone dev image build
dev-build:
	cd matrixone && docker build --platform=${PLATFORM} . -t ${IMAGE}:${TAG}

.PHONY: matrixone client start
mo-client:
	docker run -it --rm --name mo-client --network docker-compose_monet arey/mysql-client -h ${HOST} -P ${PORT} -udump -p111 
```

File: docker-compose/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Deploy MatrixOne Cluster with Docker Compose
```
Comment (sukki37) [inline docker-compose/README.md:1]: Need Improvement

Comment (sukki37) [reply docker-compose/README.md:1]: # Deployment Introduction

## Build
1. Make sure `Docker` is installed.
2. Use `make` to start/delete a 3-node MatrixOne cluster

## Usage

### Start a 3-node cluster with default configuration

You can use the following command to start a MatrixOne cluster using the official MatrixOne image (which will be automatically downloaded from the docker hub).

```shell
# Start cluster using the latest image.
make up

# Or start cluster using a specific image by specifying `tag`.
make up TAG=<TAG>
```
You can also build a MatrixOne Image from the source directly.

```
# Clone MatrixOne into the current directory, default branch is `main`.
make dev-pre

# You can also specify a branch name when executing the following command.
make dev-pre BRANCH=<BRANCH>
```
### run mo-client to connect the cluster

```shell
# connect mo cluster
make mo-client HOST=<YOUR MACHINE IP> PORT=<CLUSTER_PORT>
```

### Close cluster
```
make down
```

### Clear residual data
```
make clean
```

File: docker-compose/Makefile (hunk 1 above)
Comment (wanglei4687) [reply docker-compose/Makefile:1]: got it

**Review by sukki37 — APPROVED** [review]
LGTM

**Pull request merged**